### PR TITLE
[dev/sre/thanos_use-statefulset-for-compactor] Compactor use sts

### PR DESCRIPTION
**Pull request by pmialon**
Use statefulSet for compactor to ease operation during upgrade.

### CODE REVIEW

**Comment by Sayrus** [comment]
I don't think it makes sense to keep the name `compact-deployment.yaml` as it can now be a StatefulSet.

File: thanos/templates/compact-deployment.yaml [hunk 1]
```diff
@@ -73,12 +76,14 @@ spec:
           mountPath: /var/thanos/compact
         resources: {{ toYaml .Values.compact.resources | nindent 10 }}
       volumes:
+      {{- if no .Values.compact.statefulSet.enabled }}
```
Comment (Sayrus) [inline thanos/templates/compact-deployment.yaml:79]:
```suggestion
      {{- if not .Values.compact.statefulSet.enabled }}
```

**Comment by pmialon** [comment]
> I don't think it makes sense to keep the name `compact-deployment.yaml` as it can now be a StatefulSet.

Yes but compact.yaml is ambiguous too and I wanted to limit the diff to the maximum.

File: thanos/values.yaml [hunk 2]
```diff
@@ -406,6 +406,11 @@ compact:
   #    selector: {}
   #    volumeName: ""
   #    volumeMode: ""
+  # StatefulSet rather than Deployment
+  statefulSet:
+    enabled: false
+    size: 10G
```
[the remark this answers is not in the document]
Comment (pmialon) [reply thanos/values.yaml:412]: The usual best practice I notice in Helm Chart is to define all the keys used in the chart even if they are not used by default.
See: https://github.com/blablacar/banzai-charts/blob/master/thanos/values.yaml#L119-L131

[the remark this answers is not in the document]
Comment (Sayrus) [reply thanos/values.yaml:412]: It is useful to have size and storageClassName with a default value as it allows for a simpler `values.yaml`:
```yaml
statefulSet:
  enabled: true
```

File: thanos/values.yaml [hunk 3]
```diff
@@ -406,6 +406,11 @@ compact:
...
+    storageClassName: default
```
[the remark this answers is not in the document]
Comment (Sayrus) [reply thanos/values.yaml:413]: It is useful to have size and storageClassName with a default value as it allows for a simpler `values.yaml`:
```yaml
statefulSet:
  enabled: true
```

[the remark this answers is not in the document]
Comment (jbfavre) [reply thanos/values.yaml:413]: You're right. I misunderstood values.yaml from Chart and values from Helm Release

**Comment by pmialon** [comment]
> > > I don't think it makes sense to keep the name `compact-deployment.yaml` as it can now be a StatefulSet.
> > 
> > 
> > Yes but compact.yaml is ambiguous too and I wanted to limit the diff to the maximum.
> 
> I prefer to have a bigger cristal clear diff than an ambiguous one, and I think upstream will want that too

The file `compact-deployment.yaml` was renamed `compact.yaml`

File: thanos/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbfavre) [reply thanos/values.yaml:412]: You're right, I made a mistake mixing values.yaml from Chart and Helm Release.
My bad

**Comment by jbfavre** [comment]
lgtm

**Pull request merged**